examples/with-react-helmet/pages/_document.js

```diff
@@ -1,52 +1,35 @@
-import Document, { Head, Main, NextScript } from 'next/document'
+import Document, { Main, NextScript } from 'next/document'
 import Helmet from 'react-helmet'
 
 export default class extends Document {
   static async getInitialProps (...args) {
-    const documentProps = await super.getInitialProps(...args)
+    const documentProps = await Document.getInitialProps(...args)
     // see https://github.com/nfl/react-helmet#server-usage for more information
     // 'head' was occupied by 'renderPage().head', we cannot use it
     return { ...documentProps, helmet: Helmet.renderStatic() }
   }
 
-  // should render on <html>
-  get helmetHtmlAttrComponents () {
-    return this.props.helmet.htmlAttributes.toComponent()
-  }
+  render () {
+    const { htmlAttributes, bodyAttributes, ...helmet } = this.props.helmet
 
-  // should render on <body>
-  get helmetBodyAttrComponents () {
-    return this.props.helmet.bodyAttributes.toComponent()
-  }
+    const htmlAttrs = htmlAttributes.toComponent()
+    const bodyAttrs = bodyAttributes.toComponent()
 
-  // should render on <head>
-  get helmetHeadComponents () {
-    return Object.keys(this.props.helmet)
-      .filter(el => el !== 'htmlAttributes' && el !== 'bodyAttributes')
-      .map(el => this.props.helmet[el].toComponent())
-  }
+    // we replace next's <Head> component and build our own <head> here
+    // we could rearrange the head tags in a better order
+    // (first meta charset, then title, ...)
+    const headComponent = (
+      <head>{Object.values(helmet).map(el => el.toComponent())}</head>
+    )
 
-  get helmetJsx () {
-    return (<Helmet
-      htmlAttributes={{lang: 'en'}}
-      title='Hello next.js!'
-      meta={[
-        { name: 'viewport', content: 'width=device-width, initial-scale=1' },
-        { property: 'og:title', content: 'Hello next.js!' }
-      ]}
-    />)
-  }
-
-  render () {
-    return (<html {...this.helmetHtmlAttrComponents}>
-      <Head>
-        { this.helmetJsx }
-        { this.helmetHeadComponents }
-      </Head>
-      <body {...this.helmetBodyAttrComponents}>
-        <Main />
-        <NextScript />
-      </body>
-    </html>)
+    return (
+      <html {...htmlAttrs}>
+        {headComponent}
```

Comment:
Ah yeah 😅Forgot about that

Maybe this can be fixed by creating `class MyHead extends Head` ?

Or it could be fixed by passing helmet down to _app and using `next/head` in _app, something like this :
```jsx
<Head>{Object.values(helmet).map(el => el.toComponent())}</Head>
```

Reply:
I actually think the current implementation is fine, it's just a tradeoff between having a check for `<title>` or not, potentially we could introduce a prop to disable the check 🤔

Reply:
Ok 🙂

Maybe the only thing that should be changed is moving `helmetJsx` to `_app.js` (or to a page) instead of `_document.js`, so it's also run on the browser ?

Reply:
The way it works in the browser is different, in the browser it will use the browser api to directly inject as far as I know.